src/state/slices/foxEthSlice/foxEthSlice.ts

```diff
@@ -0,0 +1,357 @@
+import { Fetcher, Token } from '@uniswap/sdk'
+import IUniswapV2Pair from '@uniswap/v2-core/build/IUniswapV2Pair.json'
+import dayjs from 'dayjs'
+import {
+  foxAssetId,
+  foxEthLpAssetId,
+  UNISWAP_V2_WETH_FOX_POOL_ADDRESS,
+} from 'features/defi/providers/fox-eth-lp/constants'
+import farmAbi from 'features/defi/providers/fox-farming/abis/farmingAbi.json'
+import {
+  calculateAPRFromToken0,
+  getEthersProvider,
+  makeTotalLpApr,
+  rewardRatePerToken,
+} from 'plugins/foxPage/utils'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { logger } from 'lib/logger'
+import { BASE_RTK_CREATE_API_CONFIG } from 'state/apis/const'
+import { marketData } from 'state/slices/marketDataSlice/marketDataSlice'
+
+import { FOX_TOKEN_CONTRACT_ADDRESS, WETH_TOKEN_CONTRACT_ADDRESS } from './constants'
+import {
+  farmOpportunities,
+  FoxEthLpEarnOpportunityType,
+  FoxFarmingEarnOpportunityType,
+  lpOpportunity,
+} from './foxEthCommon'
+import { fetchPairData } from './utils'
+
+type FoxEthState = {
+  farmOpportunities: FoxFarmingEarnOpportunityType[]
+  lpOpportunity: FoxEthLpEarnOpportunityType
+}
+
+const initialState: FoxEthState = {
+  farmOpportunities,
+  lpOpportunity,
+}
+
+const moduleLogger = logger.child({ namespace: ['foxEthSlice'] })
+
+export const foxEth = createSlice({
+  name: 'foxEth',
+  initialState,
+  reducers: {
+    clear: () => initialState,
+    upsertLpOpportunity: (state, action: PayloadAction<Partial<FoxEthLpEarnOpportunityType>>) => {
+      state.lpOpportunity = {
+        ...state.lpOpportunity,
+        ...action.payload,
+      }
+    },
+    upsertFarmingOpportunity: (
+      state,
+      action: PayloadAction<Partial<FoxFarmingEarnOpportunityType>>,
+    ) => {
+      const stateOpportunityIndex = state.farmOpportunities.findIndex(
+        opportunity => opportunity.contractAddress === action.payload.contractAddress,
+      )
+      state.farmOpportunities[stateOpportunityIndex] = {
+        ...state.farmOpportunities[stateOpportunityIndex],
+        ...action.payload,
+      }
+    },
+  },
+})
+
+type GetFoxEthLpGeneralDataReturn = {
+  tvl: string
+  apy: string
+}
+type GetFoxEthLpWalletDataReturn = {
+  underlyingFoxAmount: string
+  underlyingEthAmount: string
+  cryptoAmount: string
+  fiatAmount: string
+}
+
+type GetFoxFarmingContractGeneralDataReturn = {
+  expired: boolean
+} & GetFoxEthLpGeneralDataReturn
+
+type GetFoxFarmingContractGeneralDataArgs = {
+  contractAddress: string
+}
+
+type GetFoxFarmingContractWalletDataReturn = {
+  cryptoAmount: string
+  fiatAmount: string
+  unclaimedRewards: string
+}
+
+type GetFoxFarmingContractWalletDataArgs = {
+  contractAddress: string
+  ethWalletAddress: string
+}
+
+type GetFoxEthLpWalletDataArgs = {
+  ethWalletAddress: string
+}
+
+export const foxEthApi = createApi({
+  reducerPath: 'foxEthApi',
+  ...BASE_RTK_CREATE_API_CONFIG,
+  endpoints: build => ({
+    getFoxEthLpGeneralData: build.query<GetFoxEthLpGeneralDataReturn, void>({
```

Comment:
Not a fan of the `general` data terminology, looking at `GetFoxEthLpGeneralDataReturn`, this is in fact representing something very specific, `tvl` and `apy`.

What about `metrics`?